addons/account/models/account.py

```diff
@@ -612,7 +612,8 @@ def onchange_price_include(self):
     def get_grouping_key(self, invoice_tax_val):
         """ Returns a string that will be used to group account.invoice.tax sharing the same properties"""
         self.ensure_one()
-        return str(invoice_tax_val['tax_id']) + '-' + str(invoice_tax_val['account_id']) + '-' + str(invoice_tax_val['account_analytic_id'])
+        return str(invoice_tax_val['tax_id']) + '-' + str(invoice_tax_val['account_id']) + '-' + str(invoice_tax_val['account_analytic_id'] or 0) \
```

Comment:
Hmm, imo if no ID is there it should always be False, so 0 is not a good alternative as the grouping key.

False or 0 will be always 0 and I prefer False over 0 even if it is stringified...
```python
>>> False or 0
0
>>> str(None)
'None'
>>> str(None or False)
'False'
```
So, if you want to avoid None as string please do the third option as it represents best a non-existent ID. But I guess it would best to find out where the None value is put instead of False as this is where the source of the issue is.
You are only fighting symptoms.